### PR TITLE
Pause queue of register payments

### PR DESCRIPTION
**Pull request by CarrilloJuan**
### What does this Pull Request do?
If the buyer doesn't have a large enough balance in Batpay, the queue of register payments will be paused
 
### Are there points in the code the reviewer needs to double check?
No.

### Why was this Pull Request needed?
To avoid a failed transaction in the blockchain and to give more information in case that the buyer doesn't have a large enough balance in Batpay

### Does this Pull Request meet the acceptance criteria?
- [x] I have read the [Contribution guidelines](https://github.com/wibsonorg/buyer-sdk/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](https://github.com/wibsonorg/buyer-sdk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: buyer-api/src/blockchain/balance.js [hunk 1]
```diff
@@ -74,14 +70,15 @@ export const hasEnoughBalance = async (address) => {
...
-  const enough = batPay.gt(amount);
+  const enough = batPay.gt(toBN(amount));
 
   if (!enough) {
     logger.error(`
     BatPay.Account '${id}' does not have enough WIB.
     Current balance: ${batPay} WIB
     Minimum balance: ${amount} WIB
     `);
+    return false;
```
Comment (Agus) [inline buyer-api/src/blockchain/balance.js]: I think you can remove this, enough would be false, and you are returning it just below 👇

File: buyer-api/src/operations/registerPayment.js [hunk 2]
```diff
@@ -29,18 +30,27 @@ export const registerPayment = async (notarizationRequestId) => {
...
   // data payload
   const {
     result: {
       notarizationFee: fee, orderId, sellers, lockingKeyHash,
     },
   } = await notarizations.fetch(notarizationRequestId);
+
   const { transactionHash, price } = await dataOrders.fetchByDxId(orderId);
+  const amount = fromWib(price);
+  if (!(await hasEnoughBatPayBalance(batPayId, amount))) {
```
Comment (nicoayala) [inline buyer-api/src/operations/registerPayment.js]:
```suggestion
  if (!(await hasEnoughBatPayBalance(batPayId, toBN(amount).muln(sellers.length)))) {
```

Comment (nicoayala) [reply buyer-api/src/operations/registerPayment.js]: I think with this change, there is no need to modify `blockchain/balance` module

File: buyer-api/src/queues/registerPaymentsQueue.js [hunk 3]
```diff
@@ -2,12 +2,15 @@ import { createQueue } from './createQueue';
...
 import logger from '../utils/logger';
 
+const queue = createQueue('RegisterPayments');
+
+const pauseQueue = async () => queue.pause();
+
 const process = async ({ id, data: { notarizationRequestId } }) => {
   logger.info(`NR[${id}] :: Process :: ${notarizationRequestId}`);
-  return registerPayment(notarizationRequestId);
+  return registerPayment(notarizationRequestId, pauseQueue, id);
```
Comment (nicoayala) [inline buyer-api/src/queues/registerPaymentsQueue.js]: I think this a really good approach to stop registering payments, let's to push it forward so the business knows little about it. What if:
```js
  return registerPayment(notarizationRequestId, (reason) => {
    logger.info(`NR[${id}] :: RegisterPayments queue paused :: ${reason}`);
    return queue.pause();
  });
```
So, inside `registerPayment` you should do:
```js
if (...condition not met...) {
  await pauseQueue('Not enough account balance ...')
}
```

**Pull request merged**